plugins/nf-tower/src/main/io/seqera/tower/plugin/BaseCommandImpl.groovy

```diff
@@ -15,6 +33,15 @@ class BaseCommandImpl {
 
     private static final int API_TIMEOUT_MS = 10_000
 
+    /**
+     * Provides common API operations for Seqera Platform
+     */
+    protected TowerCommonApi commonApi = new TowerCommonApi()
+
+    @TestOnly
+    void setCommonApi( TowerCommonApi commonApi ){
+        this.commonApi = commonApi
+    }
     /**
```

Comment:
Would not make sense to pass both TowerCommonApi and HxClient via the constructor?